staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -776,9 +779,13 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 		for property, jsonSchema := range schema.Properties {
 			subSsv := ssv
 
+			if schema.XMapType != nil && *schema.XMapType != "granular" {
```

Comment:
still would like to know why we consider atomic maps ineligible for oldSelf?

Reply:
We defined CEL transition rule placement in terms of SSA mergeability, but I can't come up with a convincing argument against supporting them on atomic map elements. Do you have any reservations about changing this in the KEP?

Reply:
I don't, but would like to hear from joe/stefan. I think oldSelf should support everything SSA merge does *at minimum*, but if there are other places it is natural/reasonable, I think it's friendly/helpful to support it there as well

Reply:
Now I'm wondering if it's not simpler to allow oldSelf in CEL rules on set lists, too, even though it's not very useful. That would leave only atomic lists as a special case.

Reply:
@jpbetz Thoughts? The phrasing in the KEP is unclear about this case, so that should be clarified either way:

> The path to the schema node being validated by a transition rule must resolve to a node that is comparable between the old object and the new object

and

> updates to descendants of collection types that are mergeable according to server-side apply may be validated by transition rules

Reply:
I don't think I have a particular objection to oldSelf access into atomic maps.

The intention of phrasing the KEP section in terms of SSA mergability was to try to highlight that mergablity is a similar to problem to determining if a oldSelf calculations can be safely performed, and that we should be able to leverage SSA mergability to more easily implement this feature.

It makes a lot of sense for atomic lists where the mergability is not safe because list[5] might have no relation to oldList[5]. But for atomic maps and structs, the mergability rule is more about whether it makes sense to be able to own and modify a field independent of the rest of the atomic stanza. And validating a field independent of the rest of a atomic stanza is very different than being able to own and modify it independent of the rest of the stanza.

I'd be okay with making the contents of atomic maps and atomics structs eligible for oldSelf calculation.  They're primarily used (in k8s native types at least) for selector types, reference types and some coordinate types (endpointAddress, endpointPort), having the ability to set a rule on a particular field seems safe in those cases. 

If no one else objects I'm in favor of making the adjustment here in this PR and then reflecting it in the KEP via an update sometime before the release.

Reply:
Thanks. Just updated the PR to allow CEL rules to reference oldSelf on atomic maps.